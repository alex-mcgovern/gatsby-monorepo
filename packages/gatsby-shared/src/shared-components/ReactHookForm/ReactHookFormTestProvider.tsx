import type { JSXElementConstructor, ReactElement } from "react";
import React, { cloneElement } from "react";
import { FormProvider, useForm } from "react-hook-form";

export function ReactHookFormTestProvider({
  children,
  ...rest
}: {
  children: ReactElement<unknown, string | JSXElementConstructor<unknown>>;
}) {
  const reactHookFormMethods = useForm();

  return (
    <FormProvider {...reactHookFormMethods}>
      {cloneElement(children, {
        ...rest,
      })}
    </FormProvider>
  );
}
