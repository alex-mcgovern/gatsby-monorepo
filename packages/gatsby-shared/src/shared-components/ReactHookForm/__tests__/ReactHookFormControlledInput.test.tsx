/** @jest-environment jsdom */
import { cleanup } from "@testing-library/react";
import React from "react";
import { renderTestComponent } from "~test/renderTestComponent";
import type { ReactHookFormControlledInputProps } from "../ReactHookFormControlledInput";
import { ReactHookFormControlledInput } from "../ReactHookFormControlledInput";
import { ReactHookFormTestProvider } from "../ReactHookFormTestProvider";

afterEach(cleanup);

/**
 * UPDATING VALUE
 */

test("Given required props, renders without errors", async () => {
  const { component } = renderTestComponent<ReactHookFormControlledInputProps>(
    <ReactHookFormTestProvider>
      <ReactHookFormControlledInput
        id="input"
        name="Test input"
        errorMessage="Error"
        label="test input"
      />
    </ReactHookFormTestProvider>
  );

  expect(component as HTMLInputElement).not.toBeNull();
});

test("Given required props, matches snapshot", async () => {
  const { component } = renderTestComponent<ReactHookFormControlledInputProps>(
    <ReactHookFormTestProvider>
      <ReactHookFormControlledInput
        id="input"
        name="Test input"
        errorMessage="Error"
        label="test input"
      />
    </ReactHookFormTestProvider>
  );

  expect(component as HTMLInputElement).toMatchSnapshot();
});

test("Given an input without value, when changing, it should have the new value", async () => {
  const { user, getByRole } =
    renderTestComponent<ReactHookFormControlledInputProps>(
      <ReactHookFormTestProvider>
        <ReactHookFormControlledInput
          id="input"
          name="Test input"
          errorMessage="Error"
          label="test input"
        />
      </ReactHookFormTestProvider>
    );

  const input = getByRole("textbox") as HTMLInputElement;

  expect(input?.value).toBe("");
  await user.type(input, "New value");
  expect(input?.value).toBe("New value");
});
