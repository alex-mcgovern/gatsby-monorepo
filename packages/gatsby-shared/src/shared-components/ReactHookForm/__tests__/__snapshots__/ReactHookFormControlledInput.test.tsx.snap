// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given required props, matches snapshot 1`] = `
<input
  class="InputBase_getInputWrapperStyles__5mrrr91 InputBase__5mrrr90 get_sprinkles_paddingTop_spacing2_mobile__1reh6ztg get_sprinkles_paddingBottom_spacing2_mobile__1reh6znp get_sprinkles_paddingLeft_spacing2_mobile__1reh6zpm get_sprinkles_paddingRight_spacing2_mobile__1reh6zrj get_sprinkles_width_100%_mobile__1reh6z107 focus_ring_styles_focusRingStyles__xbo52k0 disabled_state_styles_disabledStateStyles__uj6nll0 InputBase_getInputWrapperStyles_size_md__5mrrr93 variant_interactive_element_size_variantInteractiveElementSize_md__1dzyhqw9 variant_interactive_element_size__1dzyhqw1 get_sprinkles_fontSize_body_lg_mobile__1reh6zfg get_sprinkles_fontSize_body_sm_desktop__1reh6zfo get_sprinkles_lineHeight_body_lg_mobile__1reh6zgs get_sprinkles_lineHeight_body_sm_desktop__1reh6zh0 get_sprinkles_height_spacing6_mobile__1reh6zbd get_sprinkles_height_spacing5_desktop__1reh6zbc"
  data-testid="tested-component"
  id="input"
  name="Test input"
  value=""
/>
`;
